### ISSUE TITLE
🐛 No highlighting in vs code workspaces 

### ISSUE THREAD

**Issue by AdrianSchmidt91** [biomejs/biome-vscode#18]
### Environment information

```block
not needed
```

### What happened?

Open a workspace in vs code (for eg. an application that runs in combination with an other combination)
Highlighting does not work

Open the files without workspaces - highlighting works.

### Expected result

That highlighting is working

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by nhedger** [comment]
Went a bit fast over this last time.

I believe there to be two factors to this issue:

1. The VS Code extension checks for the presence of `biome.json` at the root of the first workspace folder to determine whether it should continue the activation process or abort it. This can be bypassed by not requiring the configuration file, but is not the reason why it does not work in multi-folder workspaces.

2. The LSP server does not have support for multiple workspace folders, and attempts to find `biome.json` at the `{root_uri}/biome.json` path. My guess is that it maps to the first available workspace folder, but I'm not certain. I'd have to do some tests.

**Comment by hulkish** [comment]
Having this issue also, in vscode it seems to just ignore my project's `biome.json` and actually causes issues (for ex, by using undesired quote type for formatting)

**Comment by atomiks** [comment]
Coming from #75:

> The LSP server does not have support for multiple workspace folders, and attempts to find biome.json at the {root_uri}/biome.json path. My guess is that it maps to the first available workspace folder, but I'm not certain. I'd have to do some tests.

I can confirm (based on my issue above) that it seems to use the first config file it finds in the list of folders in the workspace. 

I believe Prettier's extension works down-up instead: using whichever file was saved, it should traverse up the directory tree and use the first `biome.json` it finds, rather than a top-down approach from the workspace root. Are there limitations where this doesn't work?

**Comment by arendjr** [comment]
I think the way Prettier works here should be the way to indeed. That would work well with both workspaces and monorepos.

**Comment by john-wennstrom** [comment]
I get kind of the same issue. Only when I put the biome.json in the root of my monorepo project, the biome plugin take into account my defined rules.

**Comment by ematipico** [comment]
I'm going to close this issue. People can follow https://github.com/biomejs/biome/issues/1573 for updates.

The LSP doesn't support workspaces yet

**Comment by kendallroth** [comment]
> Open the files without workspaces - highlighting works.

Apologies for commenting on a closed issue, but _does_ this actually work? When attempting to use Biome in a VS Code project with no workspace configuration, I'm not getting any highlighting whatsoever, so I was a little confused to see this 🤷?

**Comment by flekschas** [comment]
I might be doing something wrong but has anybody figured out how to get highlights working when the JS project is in a subfolder?

I have a JS project under `js` in my project. Following https://deploy-preview-974--biomejs.netlify.app/reference/vscode/#defining-projects-in-a-workspace-folder I've created a file under `.vscode/settings.json` and set it to

```json
{
  "biome.projects": [
    {
      "path": "js"
    }
  ]
}
```

However, I'm not getting any highlights. I've restarted code but that doesn't help either.

I'm running
- VSCode Version: 1.91.1
- Biome Plugin Version: v2024.10.40751 (pre-release)
- Biome Version: v1.9.3
